dht_net.go

```diff
@@ -46,6 +46,9 @@ func (w *bufferedDelimitedWriter) Flush() error {
 // handleNewStream implements the inet.StreamHandler
 func (dht *IpfsDHT) handleNewStream(s inet.Stream) {
 	defer s.Reset()
+	if dht.client {
+		return
```

Comment:
This doesn’t make sense any longer, as we only attach this handlers when not in client mode. And since this goes away, so does the internal client flag.